OpenRA.Mods.Common/SpriteLoaders/PngSheetLoader.cs

```diff
@@ -141,6 +149,15 @@ void RegionsFromSlices(Png png, out List<Rectangle> regions, out List<float2> of
 
 			var framesPerRow = png.Width / frameSize.Width;
 
+			var rows = (int)Math.Ceiling((decimal)frameAmount / framesPerRow);
+			if (!new Rectangle(0, 0, png.Width, png.Height).Contains(new Rectangle(0, 0, frameSize.Width * frameAmount / rows, frameSize.Height * rows)))
```

Comment:
Same two comments here

Reply:
This isn't in a loop.

Reply:
I see. In this case we only care about the last entry, so can simplify down to checking against `png.Width` and `png.Height` directly (no need for `Rectangle.Contains()`).

Reply:
This can simplify even further: the only case where this can possibly fail is when both `FrameSize` and `FrameAmount` are defined. We can move the check up into the `if (png.EmbeddedData.ContainsKey("FrameAmount"))` branch and deal with `frameSize` and `frameAmount` directly.

Reply:
Simplified.